source/iface/error-handling-rules.rst

```diff
@@ -7,3 +7,136 @@
+  constructed with an :ref:`sycl::async_handler
+  <iface-async-error-handler>`, that handler is
+  invoked to handle the error.
+* Otherwise when no handler was passed to either :ref:`queue`
+  or :ref:`context` on construction, then a default handler
+  is invoked to handle the error. The behavior is described
+  in the :ref:`behavior-without-async_hand`.
+* All handler invocations in this list occur at times as defined
+  in the :ref:`iface-async-error-handler`.
+
+If the SYCL runtime cannot associate an asynchronous
+error with a specific queue, then:
+
+* If the context in which the error occurred was
+  constructed with an :ref:`sycl::async_handler
+  <iface-async-error-handler>`, then that handler
+  is invoked to handle the error.
+* Otherwise when no handler was passed to the
+  associated :ref:`context` on construction,
+  then a default handler is invoked to handle the error.
+  The behavior is described in the :ref:`behavior-without-async_hand`.
+* All handler invocations in this list occur at times as defined
+  in the :ref:`iface-async-error-handler`.
+
+==========================================
+Asynchronous errors with a secondary queue
+==========================================
+
+If an asynchronous error occurs when running or enqueuing
+a command group which has a secondary queue specified,
+then the command group may be enqueued to the secondary
+queue instead of the primary queue.
+The error handling in this case is also configured
+using the ``sycl::async_handler`` provided for both queues.
+
+If there is no ``sycl::async_handler`` given on any of the
+queues, then the asynchronous error handling proceeds
+through the contexts associated with the queues, and if
+they were also constructed without ``sycl::async_handlers``, then
+the default handler will be used.
+
+If the primary queue fails and there is an ``sycl::async_handler``
+given at this queue's construction, which populates the
+``sycl::exception_list`` parameter, then any errors will be added
+and can be thrown whenever the user chooses to handle those
+exceptions. Since there were errors on the primary queue and
+a secondary queue was given, then the execution of the kernel is
+re-scheduled to the secondary queue and any error reporting
+for the kernel execution on that queue is done through that
+queue, in the same way as described above. The secondary queue
+may fail as well, and the errors will be thrown if there is an
+``sycl::async_handler`` and either ``wait_and_throw()`` or ``throw()`` are
+called on that queue.
+
+If no ``sycl::async_handler`` was specified, then the one associated
+with the queue's context will be used and if the context was
+also constructed without an ``sycl::async_handler``, then the default
+handler will be used. The command group function object
+event returned by that function will be relevant to the
+queue where the kernel has been enqueued.
+
+Below is an example of catching a ``sycl::exception``
```

Comment:
Please create a new heading here, since this and following text is unrelated to the section that it's in.  Please call the section "Examples of catching synchronous errors"

Reply:
Added in the latest commit.